### PR TITLE
feat: replace `audit-app` & `bundler-audit` with `osv-detector`

### PR DESCRIPTION
**Pull request by G-Rath**
We've switched our internal auditor over to using `osv-detector` and plan to start generally switching apps over time to use it.

This is a little tricky to handle here because we typically use GitHub Actions for CI which this template doesn't configure, but I realised we can just include a dedicated GHA workflow that gets copied across as that will ensure PRs on projects will require no vulnerabilities, and then the project teams can take things from there.

### CODE REVIEW

**Comment by eoinkelly** [comment]
This is a change I'd like to live with on a few apps before we make it a default here.

**Comment by G-Rath** [comment]
@eoinkelly are you happy to live with it on any of the apps you're currently working on? I'm happy to start moving some projects over to using it within my team, but would be good to have someone on an active project living with it too.

File: .github/workflows/check-with-osv-detector.yml
Comment (eoinkelly) [inline .github/workflows/check-with-osv-detector.yml:20]: This should probably be rolled into #356 when it lands rather than being a whole separate workflow eh

Comment (G-Rath) [reply .github/workflows/check-with-osv-detector.yml:20]: yeah that is probably the way to go - we don't even need a dedicated variant, just need to axe the current `audit-app` one

**Comment by eoinkelly** [comment]
@G-Rath I'd be comfortable merging this right away if it ran osv-detector and bundle-audit. Then later we replace bundle-audit once we (maybe just me?) is a bit more confident about it being a full replacement

**Comment by G-Rath** [comment]
@eoinkelly I'm on the fence about that route because of the mixed messages it could send in running different and multiple auditors for one stack but not others, and since we've already replaced all of our auditors in CC with this.

I would like to try and have our next upcoming Rails project us this to get it in the hands of a project team, but I don't mind waiting for a bit before we land this into rails-template itself.

**Comment by eoinkelly** [comment]
Should this also install `osv-detector` for local dev? or maybe a binstub (e.g. `bin/osv-detector` which calls the real thing or tells you how to get it if it's not installed)

**Comment by G-Rath** [comment]
> `bin/osv-detector` which calls the real thing or tells you how to get it if it's not installed

ooo I like this - that'd also make it a bit easier to explore automatic/opt-in installing which might help with macOS stuff too.

**Review by jakerabid — APPROVED** [review]

**Comment by G-Rath** [comment]
While I really like the idea of having a bin stub (and said I would do that as part of this at our last guild meeting), I've decided that actually I'd like to land this as-is and then do the bin stub later since it will involve bikeshedding and I want to explore some other things like making this a workflow at the GH org level, which I think will be better by having this landed.

Similarly, I've not added logic for generating an ignore list for now because there are no unpatchable vulnerabilities right now - I'll deal with that later.

**Pull request merged**
